colcon_core/package_descriptor.py

```diff
@@ -144,11 +138,19 @@ def get_recursive_dependencies(
         return dependencies
 
     def __hash__(self):  # noqa: D105
-        return hash((self.realpath, self.type, self.name))
+        return hash((self.type, self.name))
 
     def __eq__(self, other):  # noqa: D105
-        return (self.realpath, self.type, self.name) == \
-            (other.realpath, other.type, other.name)
+        if type(self) != type(other):
+            return NotImplemented
+        if self is other:
+            return True
+        if not (self.type == other.type and self.name == other.name):
```

Comment:
Please simplify this condition to: `self.type != other.type or self.name != other.name`.